test/test_arhp.rb

```diff
@@ -86,6 +86,18 @@ def test_private_method
     assert_equal true, Pool1DbA.connection.send(:test_private_method)
   end
 
+  def test_connection_proxy_equality
+    # Refer to same underlying connection and same database
```

Comment:
Let’s add an assertion on the “refer to same underlying connection” so we discover if it breaks at some point:
```rb
    assert Pool1DbA.connection.raw_connection == Pool1DbAOther.connection.raw_connection
```